docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,118 @@
+}
+```
+
+## Determining OpenShift `Identities` for LDAP Members
+This will be done in an identical way to the mapping for [LDAP password processing](https://github.com/openshift/origin/blob/master/pkg/auth/authenticator/password/ldappassword/ldap.go#L227-243) once the requisite data is collected.
+
+```go
+// LDAPIdentityFactory creates Identites for LDAP user entries.
+type LDAPUserIdentityFactory interface {
+	IdentityFor(user *ldap.Entry) (identity userapi.Identity, err error)
+}
+```
+
+## Determining OpenShift `Users` for OpenShift `Identities`
+A new `UserIdentityMapper` will need to be created: the `DeterministicUserIdentityMapper`, which will assume that the unique identifier retrieved from the LDAP response and integrated into the OpenShift `Identity` will be deterministicically mapped to an OpenShift `User`. This is required as otherwise there is no deterministic mapping from the `Identity` created and any OpenShift `User`.
+
+```go
+// DeterministicUserIdentityMapper is a UserIdentityMapper that forces a deterministic mapping from
+// an Identity to a User.
+type DeterministicUserIdentityMapper struct {
+  // AllowIdentityCollisions determines if this UserIdentityMapper is allowed to make a mapping
+  // between an Identity provided by LDAP and a User that already is mapped to another Identity.
+  AllowIdentityCollisions bool
+}
+
+func (m *DeterministicUserIdentityMapper) UserFor(identity authapi.UserIdentityInfo) (user user.Info, err error) {
+}
+```
+
+## Populating OpenShift `Group` Records With Data From Sync Job
+Given the LDAP group UID, the corresponding OpenShift `Group` can either be found with the OpenShift `Group` name from the `LDAPGroupMapper` or created it it does not yet exist. If the group returned from the LDAP server is a first-class entry, information can be migrated to the OpenShift `Group`, but if the group is encode as attributes of first-class LDAP user entries, no information other than the name of the group can be carried over. In either case, data from the old OpenShift `Group`'s `ObjectMeta` is kept unless overridden by the update and the `Group`'s `Users` are always overriden.
+
+## Exposed Tuneables for `oadm sync-group`
+The sync command will be exposed as `oadm sync-group(s) [<group-name>] --config=<location>` and invoked like `oadm sync-group <name> --config=/ldap/serverconfig.yaml` or `oadm sync-groups --config=/ldap/serverconfig.yaml`.
+
+`--config=` determines where the config `yaml` lives.
+
+`--prune` removes OpenShift `Group` records linked to an LDAP record that no longer exists
+
+`--dry-run` would be a valuable parameter to `oadm sync groups`, returning the proposed changes without making any.
+
+`-o <json,yaml>` Format for the output of a `--dry-run` should be tuneable from human-readable, `json`, or `yaml`. The latter two formats should output in a format useable by `oc replace`.
+
+`--all-openshift` would specify all OpenShift `Groups`
+
+`--all-ldap` would specify all LDAP groups
```

Comment:
add `ldap-whitelist` for specifying an ldap whitelist file.  There's no need for a congruent `openshift-whitelist`, that can be managed by proper args.